biogeochem/EDPhysiologyMod.F90

```diff
@@ -1084,154 +1127,339 @@ subroutine phenology( currentSite, bc_in )
+             currentSite%elong_factor(ipft) = 1.
+
+          elseif ( last_flush_long_ago ) then
+             ! LEAF ON: DROUGHT DECIDUOUS TIME EXCEEDANCE
+             ! If we still haven't done budburst by end of window, then force it
+
+             ! If the status is "phen_dstat_moistoff", it means this site currently has
+             ! leaves off due to actual moisture limitations.
+             ! So we trigger bud-burst at the end of the month since
+             ! last year's bud-burst.  If this is imposed, then we set the new
+             ! status to indicate bud-burst was forced by timing
+             currentSite%dstatus(ipft)      = phen_dstat_timeon ! force budburst!
+             currentSite%dleafondate(ipft)  = model_day_int     ! record leaf on date
+             currentSite%dndaysleafon(ipft) = 0
+             currentSite%elong_factor(ipft) = 1.
+
+          elseif ( exceed_min_off_period ) then
+             ! LEAF ON: DROUGHT DECIDUOUS EXCEEDED MINIMUM OFF PERIOD
+             ! Leaves were off due to time, not really moisture, so we allow them to
+             ! flush again as soon as they exceed a minimum off time
+             ! This typically occurs in a perennially wet system.
+             currentSite%dstatus(ipft)      = phen_dstat_timeon    ! force budburst!
+             currentSite%dleafondate(ipft)  = model_day_int        ! record leaf on date
+             currentSite%dndaysleafon(ipft) = 0
+             currentSite%elong_factor(ipft) = 1.
+
+          elseif ( prolonged_on_period ) then
+             ! LEAF OFF: DROUGHT DECIDUOUS LIFESPAN
+             ! Are the leaves rouhgly at the end of their lives? If so, shed leaves 
+             ! even if it is not dry.
+             currentSite%dstatus(ipft)      = phen_dstat_timeoff    !alter status of site to 'leaves off'
+             currentSite%dleafoffdate(ipft) = model_day_int         !record leaf on date
+             currentSite%dndaysleafoff(ipft) = 0
+             currentSite%elong_factor(ipft)  = 0.
+
+          elseif ( exceed_min_on_period .and. smoist_below_threshold ) then
+             ! LEAF OFF: DROUGHT DECIDUOUS DRYNESS - if the soil gets too dry,
+             ! and the leaves have already been on a while...
+             currentSite%dstatus(ipft) = phen_dstat_moistoff     ! alter status of site to 'leaves off'
+             currentSite%dleafoffdate(ipft) = model_day_int      ! record leaf on date
+             currentSite%dndaysleafoff(ipft) = 0
+             currentSite%elong_factor(ipft)  = 0.
+          end if drought_smoist_ifelse
+
+
+       case (isemi_stress_decid)
+          !------
+          ! Semi-deciduous PFT, based on ED2.  We compare the moisture with the lower
+          ! and upper thresholds. If the moisture is in between the thresholds, we must
+          ! also check whether or not the drought is developing or regressing.
+          !------
+
+
+          ! First guess elongation factor
```

Comment:
I added the comment. The first guess is entirely based on soil water availability. But we may force elongation factors to be zero or one depending on the time since last flushing or abscising event.